diff --git a/third_party/mkl_dnn/mkldnn_v1_async.BUILD b/third_party/mkl_dnn/mkldnn_v1_async.BUILD
--- a/third_party/mkl_dnn/mkldnn_v1_async.BUILD
+++ b/third_party/mkl_dnn/mkldnn_v1_async.BUILD
@@ -243,6 +243,9 @@
 cc_library(
     name = "onednn_autogen",
-    srcs = glob(["src/cpu/x64/gemm/**/*_kern_autogen*.cpp"]),
+    srcs = select({
+        "@xla//xla/tsl:aarch64_or_cross": [],
+        "//conditions:default": glob(["src/cpu/x64/gemm/**/*_kern_autogen*.cpp"]),
+    }),
     copts = [
         "-O1",
         "-U_FORTIFY_SOURCE",
@@ -265,30 +268,38 @@
     srcs = glob(
         [
             "src/common/*.cpp",
             "src/cpu/*.cpp",
             "src/cpu/**/*.cpp",
             "src/cpu/jit_utils/**/*.cpp",
-            "src/cpu/x64/**/*.cpp",
             "src/graph/interface/*.cpp",
             "src/graph/backend/*.cpp",
             "src/graph/backend/dnnl/*.cpp",
             "src/graph/backend/fake/*.cpp",
             "src/graph/backend/dnnl/executables/*.cpp",
             "src/graph/backend/dnnl/passes/*.cpp",
             "src/graph/backend/dnnl/patterns/*.cpp",
             "src/graph/backend/dnnl/kernels/*.cpp",
             "src/graph/utils/*.cpp",
             "src/graph/utils/pm/*.cpp",
             "third_party/ittnotify/*.c",
         ],
         exclude = [
             "src/cpu/aarch64/**",
             "src/cpu/ppc64/**",
             "src/cpu/rv64/**",
-            "src/cpu/x64/gemm/**/*_kern_autogen.cpp",
+            "src/cpu/x64/**",
             "src/cpu/sycl/**",
         ],
-    ),
+    ) + select({
+        "@xla//xla/tsl:aarch64_or_cross": glob(
+            ["src/cpu/aarch64/**/*.cpp"],
+            exclude = ["src/cpu/aarch64/**/acl_*.cpp"],
+        ),
+        "//conditions:default": glob(
+            ["src/cpu/x64/**/*.cpp"],
+            exclude = ["src/cpu/x64/gemm/**/*_kern_autogen.cpp"],
+        ),
+    }),
     copts = _COPTS_LIST + [
         "-DDNNL_ENABLE_ITT_TASKS",  # Enable ITT for CPU
     ],
@@ -315,44 +323,52 @@
     srcs = glob(
         [
             "src/common/*.cpp",
             "src/cpu/*.cpp",
             "src/cpu/**/*.cpp",
             "src/common/ittnotify/*.c",
             "src/cpu/jit_utils/**/*.cpp",
-            "src/cpu/x64/**/*.cpp",
             "src/graph/interface/*.cpp",
             "src/graph/backend/*.cpp",
             "src/graph/backend/dnnl/*.cpp",
             "src/graph/backend/fake/*.cpp",
             "src/graph/backend/dnnl/passes/*.cpp",
             "src/graph/backend/dnnl/patterns/*.cpp",
             "src/graph/backend/dnnl/kernels/*.cpp",
             "src/graph/backend/dnnl/executables/*.cpp",
             "src/graph/utils/*.cpp",
             "src/graph/utils/pm/*.cpp",
             "src/xpu/*.cpp",
             "src/xpu/sycl/*.cpp",
             "src/xpu/sycl/capi/*.cpp",
             "src/xpu/ocl/*.cpp",
             "src/xpu/ocl/capi/*.cpp",
             "src/xpu/ze/*.cpp",
             "src/gpu/*.cpp",
             "src/gpu/generic/*.cpp",
             "src/gpu/intel/*.cpp",
             "src/gpu/intel/**/*.cpp",
         ],
         exclude = [
             "src/cpu/aarch64/**",
             "src/cpu/ppc64/**",
             "src/cpu/rv64/**",
-            "src/cpu/x64/gemm/**/*_kern_autogen.cpp",
+            "src/cpu/x64/**",
             "src/cpu/sycl/**",
             "src/common/ittnotify/**",
             "src/gpu/amd/**",
             "src/gpu/intel/conv/jit/v2/planner/planner_main.cpp",
             "src/gpu/nvidia/**",
         ],
-    ) + [
+    ) + select({
+        "@xla//xla/tsl:aarch64_or_cross": glob(
+            ["src/cpu/aarch64/**/*.cpp"],
+            exclude = ["src/cpu/aarch64/**/acl_*.cpp"],
+        ),
+        "//conditions:default": glob(
+            ["src/cpu/x64/**/*.cpp"],
+            exclude = ["src/cpu/x64/gemm/**/*_kern_autogen.cpp"],
+        ),
+    }) + [
         ":header_generator",
         ":kernel_list_generator",
     ],
